src/library/autodj/dlgautodj.cpp

```diff
@@ -372,6 +380,22 @@ void DlgAutoDJ::updateSelectionInfo() {
     }
 }
 
+void DlgAutoDJ::updateInfo() {
+    double duration = 0.0;
+
+    int rows = m_pAutoDJTableModel->rowCount();
+    for (int row = 0; row < rows; ++row) {
+        TrackPointer pTrack = m_pAutoDJTableModel->getTrack(m_pAutoDJTableModel->index(row, 0));
```

Comment:
Loading and instantiating a heave-weight track object just to obtain the duration is overkill and will certainly block the UI if many tracks are affected.

It somehow works for a quick and dirty prototype hack. But you shouldn't do this in production code.

Reply:
Do you have a recommendation for a workaround? 

I think the track duration will be already in the library cache.
However once we take the real duration into account, we need also the cues.

Reply:
There are multiple options depending on what information is available in this context (which I am not aware of). This is just the worst option.

Reply:
I think the best way forward is to introduce the same label we use per playlist. 
https://github.com/mixxxdj/mixxx/blob/f2d670edeb483efb066bf7f00255d01df2f0017c/src/library/trackset/playlistfeature.cpp#L133
Since Auto DJ is in fact a "hidden" playlist, we can move this function into a common base class. 

I suggest to use a "<" prefix to tell the user that the displayed time is the maximum. 

In a next step we may calculate also the real runtime for a reasonable amount of tracks only. The main use case is to see I think not more than a time < 1 h ahead, for a timed show or a visit on the rest rooms or at the bar.  
For longer times we can fall back to the original "<" display version. 

What do you think?

Reply:
> Loading and instantiating a heavy-weight track object just to obtain the duration is overkill and will certainly block the UI if many tracks are affected.

Get the duration from the database instead? _Should_ be identical if track files were not changed since last library scan.